jetty-ee10/jetty-ee10-annotations/src/main/java/org/eclipse/jetty/ee10/annotations/AnnotationParser.java

```diff
@@ -717,92 +559,66 @@ public void parse(final Set<? extends Handler> handlers, Resource r) throws Exce
         if (r == null)
             return;
 
-        if (r.exists() && r.isDirectory())
+        if (!r.exists())
+            return;
+
+        if (FileID.isJavaArchive(r.getPath()))
         {
-            parseDir(handlers, r);
+            parseJar(handlers, r);
             return;
         }
 
-        String fullname = r.toString();
-        if (fullname.endsWith(".jar"))
+        if (r.isDirectory())
         {
-            parseJar(handlers, r);
+            parseDir(handlers, r, r.getPath());
             return;
         }
 
-        if (fullname.endsWith(".class"))
+        if (FileID.isClassFile(r.getPath()))
         {
-            try (InputStream is = Files.newInputStream(r.getPath()))
-            {
-                scanClass(handlers, null, is);
-                return;
-            }
+            parseClass(handlers, null, r.getPath());
```

Comment:
This should be `parseClassFile`, otherwise you're not recording the parsed  classes.

Reply:
Closing this comment, as it's covered by the one below.